src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1344,13 +1345,22 @@ private bool OnlyWhitespaceOrCommentsAfterExtent(InternalScriptExtent extent)
 
         internal bool IsPipeContinuance(IScriptExtent extent)
         {
-            return extent.EndOffset < _script.Length && PipeContinuanceAfterExtent(extent);
+            // If the first non-whitespace & non-comment (regular or block) character following a newline is a pipe, we have
+            // pipe continuance.
+            return extent.EndOffset < _script.Length &&
+                   ContinuanceAfterExtent(extent, continuanceChar: '|', skipComment: true);
```

Comment:
This is exactly how I wanted `PipeContinuanceAfterExtent` to be refactored for additional implicit line continuance scenarios. Thank you! 👍

Reply:
I removed the `skipComment` parameter for now. It can be added back if we run into a scenario where we don't want comments in middle.